pkg/services/ngalert/api/api_ruler.go

```diff
@@ -131,10 +139,11 @@ func (srv RulerSrv) RouteGetRulesConfig(c *models.ReqContext) response.Response
 
 	configs := make(map[string]map[string]apimodels.GettableRuleGroupConfig)
 	for _, r := range q.Result {
-		namespace, err := srv.store.GetNamespaceByUID(r.NamespaceUID, c.SignedInUser.OrgId, c.SignedInUser)
+		folder, err := srv.store.GetNamespaceByUID(r.NamespaceUID, c.SignedInUser.OrgId, c.SignedInUser)
```

Comment:
For this one, I see whole bunch of others (coming from `GetFolderByUID`):

- `models.ErrFolderAccessDenied` which should be a 401
- The others coming from `toFolderError`, I would classify most of them as 400 with some of them falling into the 500 criteria.

The latter seems a bit long to tackle right now so maybe we just create an issue to tackle later?

Reply:
Regarding the 1st point I think we should return 403 (that's what we return [in any other similar case](https://github.com/grafana/grafana/blob/87c3a2b7904d8a3ece112e1189063f2c8db03223/pkg/api/folder.go#L160))
About the 2nd point, as I stated above I think we can use toFolderError:
https://github.com/grafana/grafana/blob/2ca4f3ecae97b216c063206bd3d6ba8baaaa08af/pkg/api/folder.go#L144-L172